lib/hydra/works/services/generic_file/full_text_extraction_service.rb

```diff
@@ -0,0 +1,33 @@
+module Hydra::Works
+  # Extract the full text from the content using Solr's extract handler
+  class FullTextExtractionService
+    def self.run(generic_file)
+      new(generic_file).extract
+    end
+
+    delegate :original_file, to: :@generic_file
+
+    def initialize(generic_file)
+      @generic_file = generic_file
+    end
+
+    def extract
+      uri = URI("#{connection_url}/update/extract?extractOnly=true&wt=json&extractFormat=text")
+      req = Net::HTTP.new(uri.host, uri.port)
+      resp = req.post(uri.to_s, original_file.content,           'Content-type' => "#{original_file.mime_type};charset=utf-8",
+                                                                 'Content-Length' => original_file.content.size.to_s)
+      fail "URL '#{uri}' returned code #{resp.code}" unless resp.code == '200'
+      original_file.content.rewind if original_file.content.respond_to?(:rewind)
+      JSON.parse(resp.body)[''].rstrip
+
+      # logger is not available in works, so this won't be done.
```

Comment:
It's OK if logging isn't made available in the hydra-works service.  Maybe we should keep the `rescue => e` in here, and then re-raise `e` as a `Hydra::Works::FullTextExtractionError` (which you'd need to create) that can be rescued by consumers to feed a logger. If so, you can delete all the commented out code here.